File: MiniDOMBase/DMD/Anagrafica/Fonti/Providers/IFonteProvider.cs

namespace minidom
{
    public partial class Anagrafica
    {

        /// <summary>
        /// Interfaccia implementata dagli oggetto che fungono da generatori di fonti
        /// </summary>
        /// <remarks></remarks>
        public interface IFonteProvider
        {

            /// <summary>
            /// Restituisce un array contenente i nomi delle fonti gestiti da questo oggetto
            /// </summary>
            /// <returns></returns>
            /// <remarks></remarks>
            string[] GetSupportedNames();

            /// <summary>
            /// Restituisce un array delle fonti corrispondenti al nome specificato
            /// </summary>
            /// <param name="tipo"></param>
            /// <param name="onlyValid"></param>
            /// <returns></returns>
            /// <remarks></remarks>
            IFonte[] GetItemsAsArray(string tipo, bool onlyValid = true);

                /// <summary>
            /// Restituisce la fonte in base all'ID
            /// </summary>
            /// <param name="tipo"></param>
            /// <param name="id"></param>
            /// <returns></returns>
            /// <remarks></remarks>
            IFonte GetItemById(string tipo, int id);

            /// <summary>
            /// Restituisce la fonte in base al nome
            /// </summary>
            /// <param name="tipo"></param>
            /// <param name="nome"></param>
            /// <returns></returns>
            /// <remarks></remarks>
            IFonte GetItemByName(string tipo, string nome);
        }
    }
}

File: MiniDOMBase/DMD/Anagrafica/Fonti/IFonteProvider.vb
Imports minidom
Imports minidom.Sistema
Imports minidom.Databases



Partial Public Class Anagrafica
 
    ''' <summary>
    ''' Interfaccia implementata dagli oggetto che fungono da generatori di fonti
    ''' </summary>
    ''' <remarks></remarks>
    Public Interface IFonteProvider

        ''' <summary>
        ''' Restituisce un array contenente i nomi delle fonti gestiti da questo oggetto
        ''' </summary>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Function GetSupportedNames() As String()

        ''' <summary>
        ''' Restituisce un array delle fonti corrispondenti al nome specificato
        ''' </summary>
        ''' <param name="onlyValid"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Function GetItemsAsArray(ByVal nome As String, Optional ByVal onlyValid As Boolean = True) As IFonte()

        ''' <summary>
        ''' Restituisce la fonte in base all'ID
        ''' </summary>
        ''' <param name="id"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Function GetItemById(ByVal nome As String, ByVal id As Integer) As IFonte

        ''' <summary>
        ''' Restituisce la fonte in base al nome
        ''' </summary>
        ''' <param name="name"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Function GetItemByName(ByVal nome As String, ByVal name As String) As IFonte

    End Interface

   


End Class